kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/appliers/artifacts/JvmVariantArtifacts.kt

```diff
@@ -48,9 +49,26 @@ internal class JvmVariantArtifacts(
         }
 
         fromOrigin(variantOrigin) { compilationName ->
-            compilationName !in variantConfig.sources.excludedSourceSets.get()
-                    && compilationName != "test"
+            compilationIsExcluded(compilationName, variantConfig.sources)
         }
     }
 
+    private fun compilationIsExcluded(compilationName: String, variant: KoverVariantSources): Boolean {
+        if (compilationName in variant.excludedSourceSets.get()) {
+            return false
+        }
+
+        val included = variant.excludedOtherSourceSets.get()
+        // if `test` is not included so we need to check it
+        if (compilationName == "test" && "test" !in included) {
```

Comment:
Isn't it the same condition as `included.isNotEmpty() && compilationName !in included`?

Also, you can add a test case for `"test"` source set name.

Reply:
Indeed, I was a little hasty to add a separate condition